### PR TITLE
General documentation update

### LINKED ISSUE #1716: Docs don't state that libstdc++-devel is required.
Yeah, it's probably installed on most systems.
Yeah, my system didn't have it.
Yeah, I spent an hour trying to debug this.

### LINKED ISSUE #1824: Two Clang version issues
* `docs/user/setup.rst` in the current mainline branch describes the minimum supported Clang version as 3.7.
   Travis CI runs with Clang 5.0.  Seems to me as though the two should match.

* I propose that the minimum and Travis CI version of Clang be upgraded.  According to LLVM [versions](http://llvm.org/OldNews.html)
  + Clang 3.7 was released in Sept 2015
  + Clang 5.0 was released in September 2017
   
It is now almost September 2020 and will probably be past that by the
next SN release. Clang will probably be at Clang 11 by September 2020.

I would be happier to use Clang 8, released March 2019, but can see an line of reasoning
for Clang 7, released September 2018.

The doc issue is well worth fixing, in memory of Cotton Mather.  Updating the  Clang version is not
the most critical Issue, but  good housekeeping pays off in the mid-to-long run.   

Raising the minimal supported version of Clang has the added benefit of allowing a long overdue
simplification in the Discover.scala code.  Along with other changes, the test in that code for
minimal supported version could be removed.  That would allow people to use an older Clang, unsupported-by-SN,
version if it works for them.  We would not be breaking currently successful builds.
The Discover code would discover only currently supported Clang versions.

 If given the go ahead here , I will submit a PR.  Please advise.

### PR DESCRIPTION
**Pull request by WojciechMazur**
This PR adds a general update for documentation, which includes:
* Removed implementation-specific `javalib` classes
* Added missing public API `javalib` classes
* Updated clang / llvm version requirement (with merged #2080 closes #1824)
* Ergys changes according to esentail build requirements resolves #1716 
* Updated list of supported scala version for 0.4.0 release
* Added information about Commix GC
* Added missing bindins in posixlib.rst and clib.rst

### CODE REVIEW

File: docs/user/sbt.rst [hunk 1]
```diff
@@ -119,12 +120,16 @@ Garbage collectors
    More information about the collector is available as part of the original
    `0.3.0 announcement <https://github.com/scala-native/scala-native/releases/tag/v0.3.0>`_.
 
-2. **boehm.** (default through 0.3.7)
+2. **commix.** (introduced in 0.4)
+
+   Commix is pararell mark and concurrent sweep garbage collector based on Immix
```
Comment (ekrich) [inline docs/user/sbt.rst:125]: Spelling of parallel.

**Comment by LeeTibbert** [comment]
@WojciechMazur 
To help this effort along I cloned your repository and checked out the `docs/update_docs` branch of this PR.
Unfortunately I still get a warning about `blog/interflow`.   

I have no memory of either seeing or fixing this before.  It looks like it has been around
for months/years. 

You probably did not create this error, but you touched it... But then again, now so
have I.

```
$ make html
sphinx-build -b html -d _build/doctrees   . _build/html
Running Sphinx v3.2.1
loading pickled environment... done
building [mo]: targets for 0 po files that are out of date
building [html]: targets for 8 source files that are out of date
updating environment: [config changed ('source_parsers')] 23 added, 0 changed, 0 removed
reading sources... [100%] user/testing                                          
looking for now-outdated files... none found
pickling environment... done
checking consistency... done
preparing documents... done
writing output... [100%] user/testing                                           
study_WojciechDocBuild_2020-12-30T1438-0500/docs/blog/index.rst:6: WARNING: toctree contains reference to document 'blog/interflow' that doesn't have a title: no link will be generated
study_WojciechDocBuild_2020-12-30T1438-0500/docs/blog/index.rst:6: WARNING: toctree contains reference to document 'blog/interflow' that doesn't have a title: no link will be generated
study_WojciechDocBuild_2020-12-30T1438-0500/docs/blog/index.rst:6: WARNING: toctree contains reference to document 'blog/interflow' that doesn't have a title: no link will be generated
study_WojciechDocBuild_2020-12-30T1438-0500/docs/blog/index.rst:6: WARNING: toctree contains reference to document 'blog/interflow' that doesn't have a title: no link will be generated
generating indices...  genindexdone
writing additional pages...  searchdone
copying images... [100%] contrib/compilation.png                                
copying static files... ... done
copying extra files... done
dumping search index in English (code: en)... done
dumping object inventory... done
build succeeded, 4 warnings.
```

**Comment by ekrich** [comment]
The index for the blog is called `tech` not `blog` and the error is because the blog is in markdown not rst. Everything else worked fine once I upgraded `sphinx`.

I would like to eventually see us to move to markdown using  `sbt-mdoc`, `sbt-paradox`, `sbt-site` and `sbt-ghpages` or something nicer and easier to integrate in the build.

**Comment by ekrich** [comment]
This fixes the problem @LeeTibbert was seeing - https://github.com/scala-native/scala-native/pull/2101

**Review by sjrd — APPROVED** [review]
Maybe just rebase it on the latest master to take #2101 into account.

**Pull request merged**